test/client/http_service.test.js

```diff
@@ -0,0 +1,265 @@
+const http = require('http');
+const StitchMongoFixture = require('../fixtures/stitch_mongo_fixture');
+
+import { buildClientTestHarness, extractTestFixtureDataPoints } from '../testutil';
+
+describe('With a mock server', () => {
+  const TEST_SERVER_URL = 'http://localhost:7000';
+
+  let handleRequest;
```

Comment:
@makesitgo i have one more idea to offer that might make this set of tests a little bit nicer, but i'll leave it up to you if you think it's worthwhile.
basically, we'd make a small `class` wrapper for the mock server. The class would encapsulate a `requestHandler` var that can be set using a `setRequestHandler` function. Then in your test setup code, you would just call `setRequestHandler()` with the test-specific handler function, instead of assigning `handleRequest` in each block. This makes it a lot clearer that the handler is being injected into the server for test setup purposes. The other benefit of this is that the handler function lives in the same scope as the mock server itself, and we aren't scattering the test setup state across the scope of the entire `describe` block.

Reply:
I like this ... I think can roll with it